src/redis_cmd.cc

```diff
@@ -4659,22 +4661,21 @@ class CommandFetchFile : public Commander {
               svr->GetFetchFileThreadNum();
         }
         auto start = std::chrono::high_resolution_clock::now();
-        auto fd = Engine::Storage::ReplDataManager::OpenDataFile(svr->storage_,
-                                                                 file, &file_size);
-        if (fd < 0) break;
+        auto fd = UniqueFD(Engine::Storage::ReplDataManager::OpenDataFile(svr->storage_,
+                                                                 file, &file_size));
+        if (!fd) break;
 
         // Send file size and content
         if (Util::SockSend(repl_fd, std::to_string(file_size)+CRLF).IsOK() &&
-            Util::SockSendFile(repl_fd, fd, file_size).IsOK()) {
+            Util::SockSendFile(repl_fd, *fd, file_size).IsOK()) {
           LOG(INFO) << "[replication] Succeed sending file " << file << " to "
                     << ip;
         } else {
           LOG(WARNING) << "[replication] Fail to send file " << file << " to "
                        << ip << ", error: " << strerror(errno);
-          close(fd);
           break;
         }
-        close(fd);
+        fd.Reset();
```

Comment:
How about using `fd.Close()` here, it looks more intuitive than fd.Reset()

Reply:
done